src/tags.psm1

```diff
@@ -0,0 +1,1360 @@
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition = $null,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Tag
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    if ($Tag.Id -as [int]) {
+        $Tag = $Tag.Id
+    }
+
+    # we need to use Resolve-AssetPartitionIdFromSafeguardSession from the assetpartitions.psm1 module to get the AssetPartition ID so check if the module is loaded
+    if (-not (Get-Module assetpartitions)) {
+        Import-module "$PSScriptRoot\assetpartitions.psm1"
+    }
+    $AssetPartitionId = (Resolve-AssetPartitionIdFromSafeguardSession -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId)
+    if (-not $AssetPartitionId) {
+        $AssetPartitionId = -1
+    }
+    $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags"
+
+    if (-not ($Tag -as [int])) {
+        # get tag based on name
+        $local:Tags = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$($local:RelPath)" -Parameters @{ filter = "Name ieq '$Tag'"; fields = "Id" })
+    } 
+    else {
+        # Tag ID was supplied as param, not tag name
+        if ($AssetPartitionId) {
+            # Confirm that the tag with this ID actually exists.
+            $local:Tags = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$($local:RelPath)" -Parameters @{ filter = "Id eq $Tag and AssetPartitionId eq $AssetPartitionId"; fields = "Id" })
+        }       
+    }
+    if ($local:Tags.Count -eq 0){
+        # If no tag found then return nothing
+        return
+    } else {
+        # return the ID of the tag.
+        $local:Tags[0].Id
+    }
+}
+
+
+<#
+.SYNOPSIS
+Get tag from Safeguard via the REST-API.
+
+.DESCRIPTION
+Get the tag from Safeguard. A tag can be added to Assets or Accounts.
+If the tag does not exist, this function does not return anything.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition to get tags from.
+If this value is empty, the Macrocosm partition will be used.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID to get assets from.
```

Comment:
Marking this one as well.

You have, "...to get assets from." Should be, "...to get tags from.", correct?